autopush/senderids.py

```diff
@@ -58,7 +58,10 @@ def __init__(self, args):
                 log.err("senderid_list is not a dict. Ignoring")
             else:
                 self.update(senderIDs)
-        self._refresh()
+        # set up auto-refresh for senderIDs
+        if self._use_s3:
+            self.service = TimerService(self._expry, self._refresh)
```

Comment:
It's hard for me to determine from (the docs)[https://twistedmatrix.com/documents/current/api/twisted.application.internet.TimerService.html], the source indicates that (it may be using the global reactor)[http://twistedmatrix.com/trac/browser/tags/releases/twisted-15.4.0/twisted/application/internet.py#L223], so yes, this should be added as a blocking call. My apologies, I thought it was using it's own thread loop.